src/evaluation/liveRec/StandaloneMultiEventTypeLabeler.m

```diff
@@ -0,0 +1,86 @@
+classdef StandaloneMultiEventTypeLabeler
+    % class for multi-class labeling blocks by event
+    %% -----------------------------------------------------------------------------------
+    properties (SetAccess = protected)
+        minBlockToEventRatio;
+        types;
+        labelBlockSize_s;
+    end
+    
+    %% -----------------------------------------------------------------------------------
+    methods (Abstract)
+    end
+
+    %% -----------------------------------------------------------------------------------
+    methods
+        
+        function obj = StandaloneMultiEventTypeLabeler( varargin )
+            ip = inputParser;
+            ip.addOptional( 'minBlockToEventRatio', 0.75 );
+            ip.addOptional( 'types', {{'Type1'},{'Type2'}} );
+            ip.addOptional( 'labelBlockSize_s', 0.5 );
+            ip.parse( varargin{:} );
+            obj.minBlockToEventRatio = ip.Results.minBlockToEventRatio;
+            obj.types = ip.Results.types;
+            obj.labelBlockSize_s = ip.Results.labelBlockSize_s;
+        end
+        
+        %% -------------------------------------------------------------------------------
+        function y = label( obj, blockAnnotations )
+            [activeTypes, ~, ~] = getActiveTypes( obj, blockAnnotations );
+            y = activeTypes;
+        end
+    end
+    
+    %% -----------------------------------------------------------------------------------
+    methods (Access = protected)
+        %% -------------------------------------------------------------------------------
+        
+        function eit = eventIsType( obj, typeIdx, type )
+            eit = any( strcmp( type, obj.types{typeIdx} ) );
+        end
+        %% -------------------------------------------------------------------------------
+        
+        %% -------------------------------------------------------------------------------
+        function [activeTypes, relBlockEventOverlap, srcIdxs] = getActiveTypes( obj, blockAnnotations )
+            [relBlockEventOverlap, srcIdxs] = obj.relBlockEventsOverlap( blockAnnotations );
+            activeTypes = relBlockEventOverlap >= obj.minBlockToEventRatio;
```

Comment:
Since you are not checking for "in-between-blocks" any more (blocks that overlap a bit, but not enough), all of them are later getting labeled -1 by you; whereas in the pipeline these blocks get assigned NaN and then get thrown out, because it is unclear how to label them. This may decrease your evaluated performance. I'd prefer to not evaluate these blocks, because I think both answers (1 or -1) would be ok on them.